### ISSUE TITLE
weakref.finalize.atexit assignment gives false positive missing-slot error

### ISSUE THREAD

**Issue by zzjjbb** [astral-sh/ty#4465]
### Summary

This error exists from version 0.0.74 - present.

Example
```python
import weakref

obj = object()
f = weakref.finalize(obj, lambda: None)
f.atexit = False
```

```text
> ty check ty.py
error[missing-slot]: Cannot assign to attribute `atexit`: `finalize[(), object]` has no slot or instance dictionary
 --> ty.py:5:1
  |
5 | f.atexit = False
  | ^^^^^^^^
info: Attribute `atexit` is declared but is not included in `__slots__`

Found 1 diagnostic
```

`atexit` is a writable property of type `weakref.finalize`. Not sure why it is not found, and I cannot reproduce this with a custom class with empty `__slots__` + property setter.

### Version

ty 0.0.78 (dca9f9873 2026-09-02)

**Comment by AlexWaygood** [comment]
Thanks for the report! This was actually a bug in typeshed that was fixed a couple of days ago in https://github.com/python/typeshed/pull/16333

**Comment by AlexWaygood** [comment]
I triggered an out-of-schedule sync of our vendored typeshed stubs so that this will be fixed in the next release: https://github.com/astral-sh/ruff/actions/runs/33922877628
